legal-api/src/legal_api/reports/report.py

```diff
@@ -219,14 +220,14 @@ def _set_tax_id(self, filing):
             filing['taxId'] = self._business.tax_id
 
     def _set_description(self, filing):
-        if self._business:
-            corp_type = CorpType.find_by_id(self._business.legal_type)
-            filing['entityDescription'] = corp_type.full_desc
-
-            act = {
-                Business.LegalTypes.COOP.value: 'Cooperative Association Act'
-            }  # This could be the legislation column from CorpType. Yet to discuss.
-            filing['entityAct'] = act.get(self._business.legal_type, 'Business Corporations Act')
+        legal_type = self._filing.filing_json['filing']['business'].get('legalType', 'NA')
+        corp_type = CorpType.find_by_id(legal_type)
+        filing['entityDescription'] = corp_type.full_desc
```

Comment:
We don't expect `legal_type` to be NA (just a fall back) since it is available in filing json even at the time of DRAFT and once `self._business` prop is available we replace `filing['business']` with the current value (in `_populate_business_info_to_filing` function). I had to change this to support the scenario of IA paid (by then `self._business` will not be available.